src/DynamoCore/Graph/Workspaces/LayoutExtensions.cs

```diff
@@ -200,6 +217,42 @@ internal static List<GraphLayout.Graph> DoGraphAutoLayout(this WorkspaceModel wo
 
         }
 
+        /// <summary>
+        /// If a connector has connectorPins, their edges get added to the combined graph.
+        /// </summary>
+        /// <param name="combinedGraph"></param>
+        /// <param name="connector"></param>
+        private static void AddConnectorPinEdges(GraphLayout.Graph combinedGraph, ConnectorModel connector)
+        {
+            if (connector.ConnectorPinModels.Count < 1) return;
+
+            combinedGraph.AddEdge(connector.Start.Owner.GUID, 
```

Comment:
Would you put comments of these code? If I understand correctly, these are adding edge between connector startpoint -> connectorpin, between connectorpin, connector pin to connector endpoint

Reply:
Yep- that's exactly what this code is doing.

Reply:
Added some comments for clarity.